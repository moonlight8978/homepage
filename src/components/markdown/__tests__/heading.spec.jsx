import React from 'react'
import { render } from '@testing-library/react'

import Heading from '../heading'

describe('h1', () => {
  test('renders h1 with react node', () => {
    const children = <span>Hello</span>
    const { container } = render(<Heading level={1}>{children}</Heading>)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders h1 with plaintext', () => {
    const children = 'Hello'
    const { container } = render(<Heading level={1}>{children}</Heading>)
    expect(container.firstChild).toMatchSnapshot()
  })
})

describe('h2', () => {
  test('renders h2 with react node', () => {
    const children = <span>Hello</span>
    const { container } = render(<Heading level={2}>{children}</Heading>)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders h2 with plaintext', () => {
    const children = 'Hello'
    const { container } = render(<Heading level={2}>{children}</Heading>)
    expect(container.firstChild).toMatchSnapshot()
  })
})

describe('h3', () => {
  test('renders h3 with react node', () => {
    const children = <span>Hello</span>
    const { container } = render(<Heading level={3}>{children}</Heading>)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders h3 with plaintext', () => {
    const children = 'Hello'
    const { container } = render(<Heading level={3}>{children}</Heading>)
    expect(container.firstChild).toMatchSnapshot()
  })
})

describe('h4', () => {
  test('renders h4 with react node', () => {
    const children = <span>Hello</span>
    const { container } = render(<Heading level={4}>{children}</Heading>)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders h4 with plaintext', () => {
    const children = 'Hello'
    const { container } = render(<Heading level={4}>{children}</Heading>)
    expect(container.firstChild).toMatchSnapshot()
  })
})

describe('h5', () => {
  test('renders h5 with react node', () => {
    const children = <span>Hello</span>
    const { container } = render(<Heading level={5}>{children}</Heading>)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders h5 with plaintext', () => {
    const children = 'Hello'
    const { container } = render(<Heading level={5}>{children}</Heading>)
    expect(container.firstChild).toMatchSnapshot()
  })
})

describe('h6', () => {
  test('renders h6 with react node', () => {
    const children = <span>Hello</span>
    const { container } = render(<Heading level={6}>{children}</Heading>)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders h6 with plaintext', () => {
    const children = 'Hello'
    const { container } = render(<Heading level={6}>{children}</Heading>)
    expect(container.firstChild).toMatchSnapshot()
  })
})

describe('invalid heading level', () => {
  test('renders h6 with react node', () => {
    const children = <span>Hello</span>
    const { container } = render(<Heading level="abc">{children}</Heading>)
    expect(container.firstChild).toMatchSnapshot()
  })

  test('renders h6 with plaintext', () => {
    const children = 'Hello'
    const { container } = render(<Heading level="abc">{children}</Heading>)
    expect(container.firstChild).toMatchSnapshot()
  })
})
