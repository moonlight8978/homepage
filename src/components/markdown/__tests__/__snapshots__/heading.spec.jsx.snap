// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`h1 renders h1 with plaintext 1`] = `
<h1
  class="h1"
>
  Hello
</h1>
`;

exports[`h1 renders h1 with react node 1`] = `
<h1
  class="h1"
>
  <span>
    Hello
  </span>
</h1>
`;

exports[`h2 renders h2 with plaintext 1`] = `
<h2
  class="h2"
>
  Hello
</h2>
`;

exports[`h2 renders h2 with react node 1`] = `
<h2
  class="h2"
>
  <span>
    Hello
  </span>
</h2>
`;

exports[`h3 renders h3 with plaintext 1`] = `
<h3
  class="h3"
>
  Hello
</h3>
`;

exports[`h3 renders h3 with react node 1`] = `
<h3
  class="h3"
>
  <span>
    Hello
  </span>
</h3>
`;

exports[`h4 renders h4 with plaintext 1`] = `
<h4
  class="h4"
>
  Hello
</h4>
`;

exports[`h4 renders h4 with react node 1`] = `
<h4
  class="h4"
>
  <span>
    Hello
  </span>
</h4>
`;

exports[`h5 renders h5 with plaintext 1`] = `
<h5
  class="h5"
>
  Hello
</h5>
`;

exports[`h5 renders h5 with react node 1`] = `
<h5
  class="h5"
>
  <span>
    Hello
  </span>
</h5>
`;

exports[`h6 renders h6 with plaintext 1`] = `
<h6
  class="h6"
>
  Hello
</h6>
`;

exports[`h6 renders h6 with react node 1`] = `
<h6
  class="h6"
>
  <span>
    Hello
  </span>
</h6>
`;

exports[`invalid heading level renders h6 with plaintext 1`] = `
<h6
  class="h6"
>
  Hello
</h6>
`;

exports[`invalid heading level renders h6 with react node 1`] = `
<h6
  class="h6"
>
  <span>
    Hello
  </span>
</h6>
`;
